docs/core/introduction.md

```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+  
+* [Visual Basic](../visual-basic/index.yml) uses a more verbose syntax that is closer to ordinary human language. It can be an easier language to learn for people new to programming.
 
-* Publishing an app as *self-contained* produces an executable file that includes the .NET [runtime](#sdk-and-runtimes) and [libraries](#runtime-libraries), and the application and its dependencies. Users of the application can run it on a machine that doesn't have the .NET runtime installed. Self-contained apps are platform-specific, and they can optionally be published using a form of [AOT compilation](#aot-compiler).
+## Compilation
 
-* Publishing an app as *framework-dependent* produces an executable file and binary files (*.dll* files) that include only the application itself and its dependencies. Users of the application have to separately install the .NET [runtime](#sdk-and-runtimes). The executable file is platform-specific, but the *.dll* files of framework-dependent applications are cross-platform.
+.NET apps (as written in a high-level language like C#) are compiled to an [Intermediate Language (IL)](https://en.wikipedia.org/wiki/Common_Intermediate_Language). IL is a compact code format that can be supported on any operating system or chip hardware. Most .NET apps use APIs that are supported in multiple environments, requiring only the .NET runtime to run.
 
-  You can install multiple versions of the runtime side by side to run framework-dependent apps that target different versions of the runtime. For more information, see [Target frameworks](../standard/frameworks.md).
+IL needs to be compiled to native code in order to execute on a CPU, for example Arm64 or x64. .NET supports both Ahead-Of-Time (AOT) and Just-In-Time (JIT) compilation models. For example, on Android, Linux, macOS, and Linux, JIT compilation is the default, and AOT is optional (for example, with [ReadyToRun](deploy/ready-to-run.md)). On [iOS](/xamarin/ios/), AOT is mandatory (except when running in the simulator).
 
-Executables are produced for specific target platforms, which you specify with a [runtime identifier (RID)](rid-catalog.md).
+The advantage of the JIT is that it can compile an app (unmodified) to the CPU instructions and calling conventions in a given environment, per the underlying operating system and hardware. It can also compile code at higher or lower levels of quality to enable better startup and steady-state throughput performance.
 
-For more information, see [.NET application publishing overview](deploying/index.md) and [Introduction to .NET and Docker](docker/introduction.md).
+The advantage of AOT is that it provides the best app startup and can result in smaller deployments. The primary downside is that binaries must be built for each separate deployment target (the same as any other native code).
```

Comment:
This is where I'm probably conflating things. What I meant (and didn't say) is that for CoreCLR apps, the trimmer can only trim so far. For NativeAOT, it can trim much farther (into what we think of as the runtime).

Reply:
Unless you are referring to NGEN, our AOT story doesn't really have versioning problems. Right?